contracts/Proofs.sol

```diff
@@ -38,25 +37,24 @@ contract Proofs {
     return timeout;
   }
 
-  function _end(EndId endId) internal view returns (uint256) {
-    uint256 end = ends[endId];
+  function _end(RequestId request) internal view returns (uint256) {
```

Comment:
Nitpick: my personal preference would be name the variable `requestId` to highlight its type `RequestId` as opposed to a `Request` object.

Reply:
Agreed, I'll fix this.